File: 0226-invert-binary-tree/0226-invert-binary-tree.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function invertTree(root: TreeNode | null): TreeNode | null {
    return switchTree(root);  
};

function switchTree(root: TreeNode | null): TreeNode | null {
    if (!root) {
      return null;
    }

    const tmp = switchTree(root.right);
    root.right = switchTree(root.left);
    root.left = tmp;

    return root;
};

File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        return self.switchTree(root)

    def switchTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return None

        tmp = self.switchTree(root.right)
        root.right = self.switchTree(root.left)
        root.left = tmp

        return root